fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java

```diff
@@ -691,6 +691,16 @@ private ColumnDef getColumnDef(StarRocksParser.ColumnDescContext context) {
         } else if (context.NULL() != null) {
             isAllowNull = true;
         }
+        Boolean isAutoIncrement = null;
+        if (context.AUTO_INCREMENT() != null) {
+            isAutoIncrement = true;
+        }
+        if (isAutoIncrement != null && isAllowNull != null && isAllowNull == true) {
+            throw new ParsingException("AUTO_INCREMENT column must be NOT NULL ");
+        }
```

Comment:
use pseudo to add some UT for this logic

Reply:
done